mopidy_soundcloud/soundcloud.py

```diff
@@ -382,36 +370,75 @@ def parse_track(self, data, remote_url=False):
         return Track(**track_kwargs)
 
     def _update_public_client_id(self):
-        self.public_client_id = get_client_id()
+        """ Gets a client id which can be used to stream publicly available tracks """
+
+        def get_page(url):
+            return urlopen(url).read().decode("utf-8")
+
+        public_page = get_page("https://soundcloud.com/")
+        regex_str = r"client_id=([a-zA-Z0-9]+)"
+        soundcloud_soup = BeautifulSoup(public_page, "html.parser")
+        scripts = soundcloud_soup.find_all("script", attrs={"src": True})
+        client_id = None
+        for script in scripts:
+            try:
+                regex_iter = re.finditer(regex_str, get_page(script["src"]))
+                client_id = next(regex_iter)[0].split("=")[-1]
+            except StopIteration:
+                pass
+
+        self.public_client_id = client_id
+        logger.debug(
+            f"Updated SoundCloud public client id to: {self.public_client_id}"
+        )
 
     def _get_public_stream(self, progr_stream):
         params = [("client_id", self.public_client_id)]
         return self.public_stream_client.get(progr_stream, params=params)
 
     @cache()
     def get_streamable_url(self, sharing, permalink_url, stream_url):
-        if sharing == "public" and self.public_client_id is not None:
+        progressive_urls = {"preview": "", "stream": ""}
+        if sharing == "public":
```

Comment:
Did you intend to remove `self.public_client_id is not None` from here? If `public_client_id` is None (because the scraping is broken), don't we want to skip all of this?

Reply:
`public_client_id` is updated in `__init__`. You're right, then it might be an idea to also update it here if it is None